CMakeLists.txt

```diff
@@ -1,52 +1,195 @@
+		LANGUAGES C
+)
+
+# Back-porting to PROJECT_IS_TOP_LEVEL to older cmake
+# TODO: Remove when requiring cmake >= 3.21
+if (NOT DEFINED Spglib_IS_TOP_LEVEL)
+	if (CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME)
+		set(PROJECT_IS_TOP_LEVEL ON)
+	else ()
+		set(PROJECT_IS_TOP_LEVEL OFF)
+	endif ()
+endif ()
+
+# Specify C standard
+set(CMAKE_C_STANDARD 11)
+set(CMAKE_C_STANDARD_REQUIRED ON)
+set(CMAKE_C_EXTENSIONS ON)
+
+if (NOT CMAKE_BUILD_TYPE)
+	set(CMAKE_BUILD_TYPE Release)
+endif ()
+
+#[==============================================================================================[
+#                                            Options                                            #
+]==============================================================================================]
+
+include(CMakeDependentOption)
+include(FeatureSummary)
+option(METIS_TESTS "METIS: Build unit tests" ${PROJECT_IS_TOP_LEVEL})
+option(METIS_SHARED_LIBS "METIS: Build as a shared library" ${PROJECT_IS_TOP_LEVEL})
+option(METIS_INSTALL "METIS: Install project" ${PROJECT_IS_TOP_LEVEL})
+option(METIS_OpenMP "METIS: Enable OpenMP support" OFF)
+add_feature_info(METIS_OpenMP METIS_OpenMP "OpenMP support")
+option(METIS_PROGRAMS "METIS: Build executable programs" OFF)
+add_feature_info(METIS_PROGRAMS METIS_PROGRAMS "Build executable programs")
+option(METIS_IDX64 "METIS: Build with 64 bit ints" OFF)
+add_feature_info(METIS_IDX64 METIS_IDX64 "Build with 64 bit ints")
+option(METIS_REAL64 "METIS: Build with 64 bit floats" OFF)
+add_feature_info(METIS_REAL64 METIS_REAL64 "Build with 64 bit floats")
+
+#[==============================================================================================[
+#                                     Project configuration                                     #
+]==============================================================================================]
+
+list(APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake)
+# Add some helper functions
+include(METIS)
+
+# Include basic tools
+include(FetchContent)
+if (METIS_INSTALL)
+	include(CMakePackageConfigHelpers)
+	if (UNIX)
+		include(GNUInstallDirs)
+	endif ()
+endif ()
+
+# Define basic parameters
+set(BUILD_SHARED_LIBS ${METIS_SHARED_LIBS})
+
+
+#[==============================================================================================[
+#                                       External packages                                       #
+]==============================================================================================]
+
+set(gklib_find_package_args CONFIG)
+if (METIS_OpenMP)
+	if (NOT COMPONENTS IN_LIST gklib_find_package_args)
+		list(APPEND gklib_find_package_args COMPONENTS)
+	endif ()
+	list(APPEND gklib_find_package_args OpenMP)
+endif ()
+FetchContent_Declare(GKlib
+		GIT_REPOSITORY https://github.com/LecrisUT/GKlib
+		GIT_TAG cmake/modernization
+		FIND_PACKAGE_ARGS ${gklib_find_package_args}
+)
+METIS_FetchContent_MakeAvailable(GKlib)
```

Comment:
While this is correct in principle, it excludes the possibility of using a pre-built gklib via cmake, which you spent effort on in your other PR. So I think something like this would be better:

```cmake
find_package(GKlib)
if (NOT TARGET GKlib::GKlib)
    # ......
    # FetchContent_Declare(....
endif()
```

I modified this locally and `find_package(GKlib)` works great with your other CMake.